### PR TITLE
Implement Endpoints for [Module, Section, Collection, Question]

### PR DESCRIPTION
**Pull request by marymfouda**

### CODE REVIEW

File: app/api/routes/badges.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+
+@router.get("/", response_model=BadgessOut)
+def read_Badges(
+    session: SessionDep, current_user: CurrentUser, skip: int = 0, limit: int = 100
+) -> Any:
+    """
+    Retrieve Badges.
+    """
+    if current_user.is_superuser:
```
Comment (IMperiumX) [inline app/api/routes/badges.py:19]: Please revoke superuser privileges to prevent access to any requested data.

File: app/api/routes/mazes.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+
+@router.post("/{id}/upload/")
+def upload_Mazes_file(
+    session: SessionDep,
+    current_user: CurrentUser,
+    id: int,
+    file_name: UploadFile = File(...),
+):
+    # reference: https://fastapi.tiangolo.com/reference/uploadfile/
```
Comment (IMperiumX) [inline app/api/routes/mazes.py:74]: we need to remove this upload method as it's obselete in favor of uploading to azur sotrage blobs

File: app/api/routes/profiles.py [hunk 3]
```diff
@@ -29,41 +29,41 @@ def read_Profile_by_id(session: SessionDep, profile_id: UUID) -> Any:
     raise HTTPException(status_code=404, detail="User Profile not found")
 
 
-@router.post("/", response_model=ProfileOut)
```
Comment (IMperiumX) [inline app/api/routes/profiles.py:32]: Why commenting this two (profile / setting) related endpoints

File: app/main.py [hunk 4]
```diff
@@ -33,7 +33,7 @@ def custom_generate_unique_id(route: APIRoute) -> str:
 
 app.include_router(api_router, prefix=settings.API_V1_STR)
 app.mount(
-    "/files", StaticFiles(directory="files"), "files"
```
Comment (IMperiumX) [inline app/main.py:36]: we no longer need to service static files from desk

File: app/models.py [hunk 5]
```diff
@@ -10,31 +10,38 @@ class UserBase(SQLModel):
     is_active: bool = True
     is_superuser: bool = False
     full_name: str | None = None
+    # last_solved_at : datetime | None = None
```
Comment (IMperiumX) [inline app/models.py:13]: why commenting?

File: app/models.py [hunk 6]
```diff
@@ -47,8 +54,12 @@ class User(UserBase, table=True):
     id: int | None = Field(default=None, primary_key=True)
     hashed_password: str
     profile: list["Profile"] = Relationship(back_populates="user")
-    rooms: list["Room"] = Relationship(back_populates="owner")
+    Mazess: list["Mazes"] = Relationship(back_populates="owner")
```
Comment (IMperiumX) [inline app/models.py:57]: please follow python naming convetions

File: app/models.py [hunk 7]
```diff
@@ -47,8 +54,12 @@ class User(UserBase, table=True):
...
     items: list["Item"] = Relationship(back_populates="owner")
+    Pages: list["Pages"] = Relationship(back_populates="owner")
```
Comment (IMperiumX) [inline app/models.py:59]: please follow python naming convetions

File: app/models.py [hunk 8]
```diff
@@ -67,47 +78,58 @@ class ProfileBase(SQLModel):
...
     level: int
     bio: str
+    image: str
     github_link: str
     linkedin_link: str
     personal_blog_link: str
     job: str
-    active_room: bool
-    completed_rooms: str
+    active_Mazes: bool
```
Comment (IMperiumX) [inline app/models.py:86]: please follow python naming convetions

File: app/models.py [hunk 9]
```diff
@@ -67,47 +78,58 @@ class ProfileBase(SQLModel):
...
+    completed_Mazess: str
```
Comment (IMperiumX) [inline app/models.py:87]: please follow python naming convetions

File: app/models.py [hunk 10]
```diff
@@ -67,47 +78,58 @@ class ProfileBase(SQLModel):
...
     level: int
+    image: int
     bio: str
     github_link: str
     linkedin_link: str
     personal_blog_link: str
     job: str
-    active_room: bool
-    completed_rooms: str
+    active_Mazes: bool
```
Comment (IMperiumX) [inline app/models.py:107]: please follow python naming convetions

File: app/models.py [hunk 11]
```diff
@@ -67,47 +78,58 @@ class ProfileBase(SQLModel):
...
+
+    
 
 
 class ProfileCreate(ProfileBase):
     full_name: str
     country: str
     rank: int
     level: int
+    image: int
```
Comment (IMperiumX) [inline app/models.py:101]: How can image fields contain an integer value?

File: app/models.py [hunk 12]
```diff
@@ -67,47 +78,58 @@ class ProfileBase(SQLModel):
...
     personal_blog_link: str
     job: str
-    active_room: bool
-    completed_rooms: str
+    active_Mazes: bool
+    completed_Mazess: str
     friends: str
     badges: str
-    created_rooms: str
+    created_Mazess: str
```
Comment (IMperiumX) [inline app/models.py:111]: please follow python naming convetions

File: app/models.py [hunk 13]
```diff
@@ -131,65 +153,222 @@ class ProfileOut(ProfileBase):
...
-class RoomBase(SQLModel):
+################################ Mazes ##########################
+class MazesBase(SQLModel):
     title: str
     description: str | None = None
     difficulty: int
     is_active: bool
     recommended_video: str | None = None
-    room_type: str
+    Mazes_type: str
```
Comment (IMperiumX) [inline app/models.py:163]: please follow python naming convetions

File: app/models.py [hunk 14]
```diff
@@ -131,65 +153,222 @@ class ProfileOut(ProfileBase):
...
     file_name: str = None
 
 
-class Room(RoomBase, table=True):
+class Mazes(MazesBase, table=True):
     id: int | None = Field(default=None, primary_key=True)
     title: str
     owner_id: int | None = Field(default=None, foreign_key="user.id", nullable=False)
-    owner: User | None = Relationship(back_populates="rooms")
+    owner: User | None = Relationship(back_populates="Mazess")
```
Comment (IMperiumX) [inline app/models.py:209]: Reverse lookup should be treated as an object property, following the naming conventions used elsewhere.

**Review by IMperiumX — CHANGES_REQUESTED** [review]
@marymfouda @juba0x00

**Comment by IMperiumX** [comment]
@marymfouda, ensure that all migration files are included if any changes occur to the data model.

File: app/models.py (hunk 5 above)
Comment (juba0x00) [reply app/models.py:13]: added to the Profile

File: app/api/routes/badges.py (hunk 1 above)
Comment (juba0x00) [reply app/api/routes/badges.py:19]: Do you want to remove all the checks for the superuser and make it readable by everyone? 
(for mazes and badges, ...etc.)

**Comment by marymfouda** [comment]
@IMperiumX  
احنا ليه عامليين create , delete لل profile مع ان المفروض لو في user في profile والعكس

File: app/models.py (hunk 14 above)
Comment (juba0x00) [reply app/models.py:209]: @IMperiumX 
"Reverse lookup should be treated as an object property": Explain it in detail.
naming conventions resolved in the whole project

Comment (marymfouda) [reply app/models.py:209]: Why are we use create , delete profile , It is linked to user,  when user is created, profile is created and when user is deleted, profile is deleted? @IMperiumX

**Pull request closed without merging**